### PR TITLE
TEST/EXAMPLES: explicitly set listener address; use INADDR_ANY if not provided

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
## Why ?
To test `ucp_client_server` functionality for a specific IP address.

### CODE REVIEW

**Comment by shamisp** [comment]
ok to test

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12472/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 7 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 07-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/f789b7b3253e573a35873f0c97ea6f919786dff1/_main) | :x: FAILURE
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/348e41b981288efd5a49b03c813facc00c815ece/hpc-test-node-gpu_W0) | :x: FAILURE
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/4c862580287f82a2107b7ab2b4b1827246d9282c/hpc-test-node-gpu_W3) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/359c202f4bb5a6a3fa6d024eee24847c6785312e/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/dbb4049b698bbd1dabdf0068769c38438a07101e/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/30082ff589b4316c51960beef7be7a887b3386b0/hpc-test-node-new_W0) | :x: FAILURE
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/3766999cfd46daeef143c62a987134b4ad99b9e7/hpc-test-node-new_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/c82a3d21f10b3cdb30b64be63874d3c1aec4851d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/ae5d8a0a9662d4356bc5116c141734477571d404/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/0623b0fc17e4545aa014aabf3967128d0a261704/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/6dba225de2ad822d0f83fdd61076648c7f92b069/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/96e98cf23a68eb90219e130e948e0194ef2dd592/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/5e2b19f12c45b12779bafce97ac65b448db69a43/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/b3c1af21a7019ff758ef2b7375da6da8e1bd9207/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/5861f281210fb4b7b25359f963dec277821d5733/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/a0ed14df967ede6c81d5c219089f736e7f5740b3/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/3d52f9b703ce057a6f4cad211fd48ba23bc47a88/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/d5eac3fe07c9672190d5761e6344810276aeb35b/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/62463809f923a706408942683186b170ae6bdcd3/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/a539cc590cb6a10bb30d835abf06d69be8ec6a01/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/cd7e3ba42d916ea634e538eec3fd457001fb5134/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/ea3fb774ea9de29dc90f9157d143a4c8c2e0c24b/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/7db247c23c2276c4fb7372a2847e9dc3bb755cf4/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/a5ca874dc296866e72159aa1cceb968d61974639/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/3ee785d81624944e530b7e1028a2ea399d349dfe/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/bb975aa2cae2552d04e11259e658edb7c622e41c/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/31ab39acb893c52ba9bb48a9e9e06dd2c5201628/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/f11b36fdd91830fea9fa2826610bb57d02cf785d/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0f8523aedc846f34441f7eb6bf540e53/raw/b76deab3a13cfbd00a270549d8d42228e091b4cd/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 7 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 07-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/44e6751ef31bdf9ad6fb9e521cf34d9f9d952519/_main) | :x: FAILURE
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/16d5ecf229948fa82af5ad4319ad5f1f3d40f2f7/hpc-test-node-gpu_W0) | :x: FAILURE
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/23a607d4c544304612850a73b38f66e415e35b7f/hpc-test-node-gpu_W3) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/e5ea7095164cbdb3b16afe17b66d49f32ece2119/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/bac7dd671b50a9886fab3c449897c8f3795d4e86/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/8b45f73fb1534561a859d502f4887e9d69d03a7f/hpc-test-node-new_W0) | :x: FAILURE
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/adfaa4484096fa4ae9b9872d8258cfeb6da0ef30/hpc-test-node-new_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/a73dc34dca461454876c43305a938818e52eeece/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/956a90fa8d0169e6dd50c3f3ccd04ea9a984dc99/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/f713b5ba4ef341fd58a6d0b2376487055987f874/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/b679013cc5ccefacc590ddf7e3e9b26d819dcaed/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/c4c36c610481266bab4ee4b6f362aad10ca95f32/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/ae7c0d473a9dab08f30c37441e0937ad682b0091/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/5ca6f008634d3ff55b765f1ac8e21394a750b580/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/668c80af0b9ecd030915e32a42b11d89ea7180f8/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/07ef16e039459e75e7b50705fa8c1d48578c6242/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/d8f4faf5da264c6087299aaf26f5ee6e523fc831/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/ccf0db497fa3f84a58479359641f9c83cd0a69ec/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/94f3fa03624566b5bcea4183301e7924be4304b1/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/5a7586df4cc8e42757f9696ec7c1360a03cbc5c1/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/dfc559222358ac70b18d798adbd8f696df468938/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/673ecf0a273536878e2cf7e5b7515d669e265a96/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/1ebdc473c70457a99d2dd6e5372afaad396a747f/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/114b0e774ee6ce6eb5ede313c9ea450424cb8cd7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/1c3b31d119edd127c1d0da70aa82601a6ca91192/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/f8172ce1d1006c1e681a80c944a2586f12147078/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/26fead1651b06b6c4b96fb97f6a47fefab4f4e08/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/a1e62fd565a58ff9a03028762f6e96d47f34219f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9e7a56b7c27efd5253b8b5dbd4020ad1/raw/fe722a7733577b6921620306b808f1247fca7a49/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by Akshay-Venkatesh** [comment]
Hi @shamisp Looks like a license expiry issue for the errors:

```

360 C/C++ compilation units (100%) are ready for analysis
The cov-build utility completed successfully.
+ cov-analyze --enable-fnptr --fnptr-models --all --paths 20000 --security --concurrency --dir /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node-legacy/worker/3/cov_build_release_12473
Coverity Static Analysis version 8.5.0 on Linux 3.10.0-693.el7.x86_64 x86_64
Internal version numbers: c8d197a567 p-kent-push-26368.915

[FATAL] License authorization failure: License has expired.
```

I don't see other issues as such.

File: test/examples/ucp_client_server.c [hunk 1]
```diff
@@ -101,12 +101,12 @@ static void err_cb(void *arg, ucp_ep_h ep, ucs_status_t status)
...
  * Set an address for the server to listen on - INADDR_ANY on a well known port.
  */
-void set_listen_addr(struct sockaddr_in *listen_addr)
+void set_listen_addr(const char *address_str, struct sockaddr_in *listen_addr)
 {
     /* The server will listen on INADDR_ANY */
     memset(listen_addr, 0, sizeof(struct sockaddr_in));
     listen_addr->sin_family      = AF_INET;
-    listen_addr->sin_addr.s_addr = INADDR_ANY;
+    listen_addr->sin_addr.s_addr = (address_str) ? inet_addr(address_str) : INADDR_ANY;
```
Comment (evgeny-leksikov) [inline test/examples/ucp_client_server.c]: should we handle IPv6? IMO we have to check sa_family at least

**Comment by shamisp** [comment]
I recently upgraded arm compiler on one of machines. Let me double check what is going on.

**Comment by shamisp** [comment]
> I recently upgraded arm compiler on one of machines. Let me double check what is going on.

Okay, seems like it is unrelated to arm compiler. @mike-dubman can you please take a loot at the license issue ?

**Comment by mike-dubman** [comment]
license fixed by @MrBr-github

**Comment by brminich** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12498/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 10-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/956567c45ca4a8b69a0c70c272bfde0183eecc22/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/eec135e77ff12897150b39856d8ba56563c5d471/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/65c997881b1bad86ae3a2f06716651f914a46b83/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/a6944b1ba37f5d18b07d9e8e350a5323915327a9/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/bfd1a77d520d08e5bb6ceb0a2d46bcb3b696a52d/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/fdc1f462de0c2991aab6c0dcced1adfd31ef504e/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/74d8b65eb8305ef2bcc9522edfc66447199e125f/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/6a6f370b5d552b254c49ec8ca07e4046530dfe02/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/8e9c5584588409d049a71d84931d0a0b844c5348/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/a625bf3e62540673d1b937b789939fd2c722b755/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/367fa87a5ca67c9806d230d992a2b63a4599d1f0/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/83d03515dcc719a8e12094d17ca30282b9747300/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/a7c84e447ea6c266578f273edbc02a95d8c71c9c/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/d4a59eff2f30655a1bf3cac7b6097d28e9cff5fb/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/d2fd4dd41aeaa076354d131711a5867b426a306b/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/8463ac3c2b4abead8cf25e20d9714f97da7147dd/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/764a28e22c7cbe28d207d352bd0463c17bfe1fb0/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/f9c29366747dfd7c51f8a842f17c4e93fef4b56b/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/c1b95111db8571d2657baf9f0ac03409f29e928e/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/646662866cfde99cb0e8587bab40588929d069c1/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/4fdd63834d59d140010ffbd3d157e282831cba28/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/eb06bd9313fff53e4653d4dab407eee7acfa8993/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/0b1a3567a94ff1e9e0e881bb702945504a9e1caf/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/6ef6512d25d48ae9b3cb47c12960645f744b98df/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/a33ab8adeadea8c70e53cb490c126b933d109a8d/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/28a27d254a9d7db4de7fe0e81570c8f1d29780b4/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/21299b7f4636235a63c5710e4f802eef0c0b15cf/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/785d6998dbb62461f374a199c135ed651d23bf57/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/683e8ccdf7c80ac111204dc8578a1723/raw/1f5f2615bb4b64e1be57c07d9db0543a55d5ee3e/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: test/examples/ucp_client_server.c (hunk 1 above)
Comment (Akshay-Venkatesh) [reply test/examples/ucp_client_server.c]: @evgeny-leksikov I'm in favor of this. Should we handle it in this PR or a follow-up? Either works for me. Lmk

Comment (evgeny-leksikov) [reply test/examples/ucp_client_server.c]: @Akshay-Venkatesh up to you. As far as I understand, in current state, there is a bug if pass IPv6 address, right?

Comment (alinask) [reply test/examples/ucp_client_server.c]: @Akshay-Venkatesh IMHO it's better to handle both case in this PR.

Comment (Akshay-Venkatesh) [reply test/examples/ucp_client_server.c]: Sorry for the late reply I was OOTO. I'll add code to handle both cases soon.

**Pull request merged**